### DISCUSSION TITLE
Under graphicall session, what are the backend GUI used ? are based on the type of desktop?

### DISCUSSION THREAD

**Discussion by mckaygerhard** [BashGui/easybashgui#16] [Q&A] [answered]
i mean, by example i runs GNOMEpuah so then will use zenity? (it makes sense) or under KDE will use kdialog?

but what if we used LXQT .. as i know there is qdialog but? will use it?

and what if we are under LXDE or worse, only under openbox or JWM that are not desktops per se?

and the worse.. what about if we used WAYLANshit? (in this case i guess must be zenity cos as i know zenity i close to gnome and of course will have good wayland support)

**Comment by mckaygerhard** [comment, marked as answer]
ok got the code at line https://github.com/BashGui/easybashgui/blob/5cb63969e87795e27e69202ef5b7f975db14821b/lib/easybashgui.lib#L386

the code it not finished yet.. only tries to detect if the kde bin package its running.. and is not based on the `env` variables..interesting.. ok that will be improved later in future releases so based on such lines i concluded current status of the detection:

1. if all the required backends are available or at least kdialog are.. the EBG will try to check if kdebin is running and only will use kdialog if the kdebin its running..
2. if only GTK based are running, the EBG will just use zenity, event if xdialog is available and there is no gnome runing (window managers or similar.. ) that is enough cos zenity seems supports waylandshit

This is simple and enough but is not optimal neither right way cos into kde we can change the window manager to another like openbox..

**Comment by vaisarger** [comment]
Hi @mckaygerhard 

Ok. Let's take a look at EBG philosophy.
1) First of all, user has not to be concerned about wich widget is in its system.
2) There is a priority list in various widgets, it depends on features, update frequency, compliance. Of course that was an author choice, and of course can be questioned, but that's it...
Priority list is:
Yad
Gtkdialog
Zenity
Kdialog
Xdialog
dialog
Only in case KDE (and its window manager ) are running, I left to user desktop coherence using Kdialog; but in all other cases Kdialog is not first choice because objectively Kdialog is a little limited.
3) All this has a bypass through "supermode" variable.

_This is simple and enough but is not optimal neither right way cos into kde we can change the window manager to another like openbox.._
In *that* case @mckaygerhard , it means (for me ) that user does not give value to KDE *total* environment, and that's reason enough for me to make EBG use Yad, or Gtkdialog, or Zenity if present in system.

**Reply by mckaygerhard** [reply]
yeah.. i noted in the code as i answered myselft some hours ago.. thanks for reply @vaisarger so is enought for gui management.. i will provide code examples for auto detectin later.. so do not disturb with simplicity of the upstream project..
